feat: ✨ sd headline

Vahid1919 opened the pull request.

Vahid1919 commented:
@solid-design-system/sds-team  Requesting your review :)

yoezlem commented:
@Vahid1919 whats the different between size 4xl and default?
<img width="1024" alt="image" src="https://github.com/solid-design-system/solid/assets/98025014/1261be93-9c92-4b26-b1a0-ba23fd53a0c9">

It seems there are no differences here, so I assume the default state is 4xl. If that's the case, please remove the 'sd-headline--size-4xl' class. This way, we would eliminate a redundant class 🙂

mariohamann commented:
Please have a look how @MarcMatthiae wrote the samples for `sd-table-cell` and adapt.

<img width="1552" alt="CleanShot 2023-09-27 at 14 53 40@2x" src="https://github.com/solid-design-system/solid/assets/26542182/d8f4d8b7-f463-4fd8-8091-d4ea5006bcf9">


Only one single `samples` Story and then something dividing them.

yoezlem commented:
<img width="1048" alt="image" src="https://github.com/solid-design-system/solid/assets/98025014/acf2e0fa-fc34-4243-ae25-e3740d4b8f96">

I'm not entirely sure if this class makes sense in this context. I understand that we want to list all attributes for users, but it can be confusing when changing the attribute to true has no effect. Could we consider adding an anchor in the description to the samples here? Or would that be too excessive?
@mariohamann what are you thinking about that?

yoezlem commented on packages/components/src/styles/headline/headline.css:
```diff
@@ -0,0 +1,79 @@
+    @apply text-[3rem] text-primary shrink-0 sm:mt-2 align-middle;
+  }
+
+  &--size-4xl {
+    @apply text-3xl sm:text-4xl;
+  }
```
https://github.com/solid-design-system/solid/pull/435#issuecomment-1737340284

yoezlem commented on packages/components/src/styles/headline/headline.declaration.ts:
```diff
@@ -0,0 +1,23 @@
+
+export default {
+  styleName: 'sd-headline',
+  summary: 'Leadtext is used for text that should be highlighted and a focal point of the page.',
```
Wrong summary..right?

Vahid1919 replied:
Yes. Will be fixed.

yoezlem commented on packages/components/src/styles/headline/headline.stories.ts:
```diff
@@ -0,0 +1,224 @@
+
+/**
+ * Headlines are vital for displaying content hierarchy and to improve accessibility. <br>
+ * A headline can be additionally accompanied by an icon. The icon can be displayed on the left side or inline. <br>
```
Please add hier also the sizes like in the other styles:
<img width="671" alt="image" src="https://github.com/solid-design-system/solid/assets/98025014/351dc7f9-a207-4218-8ea6-585a2d572ddd">

<img width="1093" alt="image" src="https://github.com/solid-design-system/solid/assets/98025014/3052a8ee-d1bc-488a-a809-fc8f04e92b23">

<img width="1048" alt="image" src="https://github.com/solid-design-system/solid/assets/98025014/3e2491e5-01a5-40e5-a537-8a82fff17668">

yoezlem commented on packages/components/src/styles/headline/headline.stories.ts:
```diff
@@ -0,0 +1,224 @@
+  argTypes,
+  decorators: [
+    (story: () => typeof html) => html`
+      <style>
```
What exactly did you insert this for? What does this <style>?

Vahid1919 replied:
I needed it to align the headline text to the left in order to highlight the inline behavior more.

Vahid1919 commented:
> @Vahid1919 whats the different between size 4xl and default? <img alt="image" width="1024" src="https://user-images.githubusercontent.com/98025014/270983283-1261be93-9c92-4b26-b1a0-ba23fd53a0c9.png">
> 
> It seems there are no differences here, so I assume the default state is 4xl. If that's the case, please remove the 'sd-headline--size-4xl' class. This way, we would eliminate a redundant class 🙂



> Please have a look how @MarcMatthiae wrote the samples for `sd-table-cell` and adapt.
> 
> <img alt="CleanShot 2023-09-27 at 14 53 40@2x" width="1552" src="https://user-images.githubusercontent.com/26542182/270984409-d8f4d8b7-f463-4fd8-8091-d4ea5006bcf9.png">
> Only one single `samples` Story and then something dividing them.

@mariohamann @yoezlem Both these reviews are addressed in the latest commits. I reacted to the "resolved" ones with 🚀 to make them easier to notice since those don't have a resolve button.

mariohamann commented:
@Vahid1919 I justa ccpeted the denied story. :)

mariohamann commented:
Removed myself from assignees – feel free to re-add, if you have changes.

yoezlem commented:
> <img alt="image" width="1048" src="https://user-images.githubusercontent.com/98025014/270984794-acf2e0fa-fc34-4243-ae25-e3740d4b8f96.png">
> I'm not entirely sure if this class makes sense in this context. I understand that we want to list all attributes for users, but it can be confusing when changing the attribute to true has no effect. Could we consider adding an anchor in the description to the samples here? Or would that be too excessive? @mariohamann what are you thinking about that?

@mariohamann

mariohamann commented:
@yoezlem I answered it and it is already fixed:

![Uploading CleanShot 2023-10-04 at 14.21.33@2x.png…]()

yoezlem commented:
> @yoezlem I answered it and it is already fixed:
> 
> <img alt="CleanShot 2023-10-04 at 14 21 33@2x" width="1088" src="https://user-images.githubusercontent.com/26542182/272571981-af2e1b68-c76c-4ef6-a07b-3cd8a95ca91f.png">

No it was no longer about the sizes but rather about the boolean for 'inline.

mariohamann commented:
Aaaah, didn't understand that from the screenshots, sorry.

<img width="1116" alt="CleanShot 2023-10-04 at 14 34 25@2x" src="https://github.com/solid-design-system/solid/assets/26542182/e6710d30-b606-47a9-a06d-b1f4f6a16226">

We have something like this in the navigation item, here.

I think explicitly telling, that this will only apply changes if there is an `sd-icon` set is fine. Maybe the documentation for that should be optimized @Vahid1919

Vahid1919 commented:
@mariohamann I added the changes to the size documentation. 

I also updated the documentation for `inline`. But I wanted clarification for what you wanted me to link?

yoezlem commented:
> > @mariohamann I added the changes to the size documentation.
> > I also updated the documentation for `inline`. But I wanted clarification for what you wanted me to link?
> 
> @Vahid1919 We wanted to know if its possible to add an Anchor in the descriptions to the samples? Could be less confusing for the users when changing the attribute to true and sees no effect, so the idea was to add an anchor to the samples to make it easier for the user?

That would be the anchor: https://64469bd84c5ea1ce2dd5d016-yexwmzbaoo.chromatic.com/?path=/docs/styles-sd-headline--docs#inline

karlbaumhauer commented on packages/components/src/styles/headline/headline.declaration.ts:
```diff
@@ -0,0 +1,24 @@
+    },
+    {
+      name: 'sd-headline--inline',
+      description: 'Sets inline behavior. Used exclusively when an sd-icon or other component is present.'
```
what other components should be used with the headline using `inline`? If we dont have one, I'd suggest to remove the last part of the sentence to prevent people having this idea...

Vahid1919 replied:
sd-mark for example (see headline Samples). Setting inline here maintains the text wrapping for smaller screens. I chose to refer to components in general since inline can be pretty useful when a developer wants to combine headline with anything else.

karlbaumhauer replied:
ok. Just asking as this might conflict with the expected usage from design. If they are fine with it, you can resolve my comment.

karlbaumhauer commented on packages/components/src/styles/headline/headline.stories.ts:
```diff
@@ -0,0 +1,278 @@
+ * <br>
+ * <b>Sizes</b><br>
+ * <li>4xl is the default size.</li>
+ * <li>3xl, xl, lg and base can be used as alternatives to 4xl.</li>
```
```suggestion
 * <li>3xl, xl, lg and base can be used to change the visual size of the HTML element.</li>
```

karlbaumhauer commented on packages/components/src/styles/headline/headline.stories.ts:
```diff
@@ -0,0 +1,278 @@
+};
+
+/**
+ * The `inline` class can be used to maintain inline positioning when used together with an Icon or other components.
```
if changed in the declaration, pls also change here.

karlbaumhauer commented on packages/components/src/styles/headline/headline.stories.ts:
```diff
@@ -0,0 +1,278 @@
+  render: (args: any) => {
+    return html` ${[
+      {
+        title: 'without an Icon',
```
```suggestion
        title: 'Without an sd-icon',
```

karlbaumhauer commented on packages/components/src/styles/headline/headline.stories.ts:
```diff
@@ -0,0 +1,278 @@
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do magna aliqua.`
+      },
+      {
+        title: 'with an Icon',
```
```suggestion
        title: 'With an sd-icon',
```

Vahid1919 commented:
@van-nguyen-ht Can you take a look at the UI Tests when you have time? :)

abudd1094 reviewed (approved):
"Headlines: Where brevity meets boldness, and a few words pack a punch louder than a thousand screams."

LGTM

Vahid1919 commented:
> > > @mariohamann I added the changes to the size documentation.
> > > I also updated the documentation for `inline`. But I wanted clarification for what you wanted me to link?
> > 
> > 
> > @Vahid1919 We wanted to know if its possible to add an Anchor in the descriptions to the samples? Could be less confusing for the users when changing the attribute to true and sees no effect, so the idea was to add an anchor to the samples to make it easier for the user?
> 
> That would be the anchor: https://64469bd84c5ea1ce2dd5d016-yexwmzbaoo.chromatic.com/?path=/docs/styles-sd-headline--docs#inline
> 
> (of course you need to take the solid-design-system.fe.union-investment.de/..... URL)

@yoezlem I added the anchor link that points to "#inline" id.  But it still reloads before scrolling to the Inline section. It opens an iframe? (the link changes from ?path to ?iframe). I would appreciate if you took a look at it :)
